src/DrawManager.sol

```diff
@@ -270,121 +309,138 @@ contract DrawManager {
-    UD2x18[] memory rewardFractions = new UD2x18[](2);
-    uint rewardPool = _totalReserve > maxRewards ? maxRewards : _totalReserve;
-    rewardFractions[0] = computeStartDrawRewardFraction(_startDrawElapsedTime);
-    rewardFractions[1] = computeFinishDrawRewardFraction(_finishDrawElapsedTime);
-
-    uint totalRewards;
-    (rewards, totalRewards) = RewardLib.rewards(
-      rewardFractions,
-      rewardPool
-    );
-    remainingReserve = _totalReserve - totalRewards;
+  function getStartDrawAuction(uint256 _index) external view returns (StartDrawAuction memory) {
+    return _startDrawAuctions[_index];
+  }
+
+  function _computeFinishDrawReward(
+    uint256 _auctionOpenedAt,
+    uint256 _auctionAt,
+    uint256 _availableRewards
+  ) internal view returns (uint256 reward, UD2x18 fraction) {
+    fraction = computeFinishDrawRewardFraction(_computeElapsedTime(_auctionAt, _auctionOpenedAt));
+    reward = RewardLib.reward(fraction, _availableRewards);
+  }
+
+  function _computeStartDrawRewards(
+    uint256 _firstAuctionOpenedAt,
+    uint256 _lastAuctionAt,
+    uint256 _availableRewards
+  ) internal view returns (uint256[] memory rewards, UD2x18[] memory fractions) {
+    uint length = _startDrawAuctions.length;
+    rewards = new uint256[](length+1);
+    fractions = new UD2x18[](length+1);
+    uint256 previousStartTime = _firstAuctionOpenedAt;
+    for (uint i = 0; i < rewards.length; i++) {
+      uint closedAt = i < length ? _startDrawAuctions[i].closedAt : _lastAuctionAt;
+      fractions[i] = computeStartDrawRewardFraction(_computeElapsedTime(closedAt, previousStartTime));
+      rewards[i] = RewardLib.reward(fractions[i], _availableRewards);
+      previousStartTime = closedAt;
+    }
   }
 
   /// @notice Computes the start draw reward.
   /// @param _startDrawElapsedTime The elapsed time between draw close and startDraw()
   /// @param _totalReserve The total reserve available to allocate rewards from
   /// @return reward The computed reward for start draw
-  function computeStartDrawReward(uint48 _startDrawElapsedTime, uint256 _totalReserve) public view returns (uint256) {
-    (uint256[] memory rewards,) = computeRewards(_startDrawElapsedTime, 0, _totalReserve);
-    return rewards[0];
-  }
+  // function computeStartDrawReward(uint48 _startDrawElapsedTime, uint256 _totalReserve) public view returns (uint256) {
+  //   (uint256[] memory rewards,) = computeRewards(_startDrawElapsedTime, 0, _totalReserve);
+  //   return rewards[0];
+  // }
 
   /// @notice Computes the reward fraction for the start draw auction.
   /// @param _elapsedTime The elapsed time since the draw closed in seconds
-  /// @return The computed reward fraction for the start draw auction
+  /// @return The computed reward fraction for the start draw auctino
   function computeStartDrawRewardFraction(uint48 _elapsedTime) public view returns (UD2x18) {
```

Comment:
Do we need these public fraction functions anymore? I feel like they are unnecessary since we have the actual reward amounts. We might want to make these internal to avoid confusion.